src/accounts/index.js

```diff
@@ -0,0 +1,110 @@
+import _ from 'lodash';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+import db from '../../models';
+import responseHelper from '../helpers/response';
+
+dotenv.config();
+
+class AccountController {
+  static async createAccount(req, res) {
+    const payload = req.body;
+
+    try {
+      let userDetails = {};
+      let accountDetails = {};
+      let isAccountCreated = null;
+      let isUserCreated = null;
+      let numOfAffectedRows = 0;
+
+      [userDetails, isUserCreated] = await db.User.findOrCreate({
+        where: {
+          firstName: payload.firstName,
+          lastName: payload.lastName,
+        },
+        defaults: {
+          membershipNo: req.body.membershipNo,
+          password: req.body.password,
+          dateOfBirth: req.body.dateOfBirth,
+          middleName: req.body.middleName,
+          email: req.body.email,
+          sex: req.body.sex,
+          phoneNumber: req.body.phoneNumber,
+          residentialAddress: req.body.phoneNumber,
+        },
+      });
+
+
+      if (!isUserCreated) {
+        [numOfAffectedRows, userDetails] = await db.User.update(
+          {
+            middleName: req.body.middleName || userDetails.middleName,
+            password: req.body.password || userDetails.password,
+            membershipNo: req.body.membershipNo || userDetails.membershipNo,
```

Comment:
First, I think this endpoint should be renamed to something else like `register` for easier comprehension. I initially thought this endpoint would only deal with only properties related to the accounts model, but then I started seeing other properties.
Second, users cannot provide `membershipNo` during registration because it's would auto-generated.

Reply:
@tijesunimi-peters for auto generating membershipNo, can you please ask @omorilewa  because I followed her direction. So @omorilewa would be in the best position to state this

Reply:
@albert85 The story does not specifically state whether or not the membershipNo would be generated. I assumed you understood what we are trying to achieve. 

If it is too much work we can split out the generation of membership number into another story.

Reply:
what suppose to generate the memberNo @omorilewa